node_lambnik/src/tiler/src/util/build-xml.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * This script uses carto to compile templated MML files
+ * to XML that mapnik can understand
+ */
+
+import carto from 'carto'
+import path from 'path'
+
+import { readFile, writeFile } from './fs-promise'
+
+const IN_FILE = process.argv[2]
+const OUT_FILE = process.argv[3]
+const NODE_ENV = process.env.NODE_ENV === 'production'
+console.log(`Production mode=${NODE_ENV}`)
+
+// Takes in a string and templates in the proper env variables
+const fillTemplate = (mmlString) => {
+    return mmlString.replace(/\$\{[a-z0-9_]+\}/gi, (match) => {
+        const varName = `${(NODE_ENV) ? 'PROD_' : 'DEV_'}${match.slice(2, match.length -1)}`
+        return `"${process.env[varName]}"`
+    })
+}
+
+// Loads properly formatted MML into an MML object
+const mmlToJSON = (fullMML) => {
```

Comment:
This isn't converting MML to JSON, it's just converting an MML string to an MML object.  `loadMML` might be a better name.